datafusion-jit/src/ast.rs

```diff
@@ -0,0 +1,359 @@
+        self.fmt_ident(0, f)?;
+        Ok(())
+    }
+}
+
+impl Display for Expr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Expr::Literal(l) => write!(f, "{}", l),
+            Expr::Identifier(name, _) => write!(f, "{}", name),
+            Expr::Binary(be) => write!(f, "{}", be),
+            Expr::Call(name, exprs, _) => {
+                write!(
+                    f,
+                    "{}({})",
+                    name,
+                    exprs
+                        .iter()
+                        .map(|e| e.to_string())
+                        .collect::<Vec<_>>()
+                        .join(", ")
+                )
+            }
+        }
+    }
+}
+
+impl Display for Literal {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Literal::Parsing(str, _) => write!(f, "{}", str),
+            Literal::Typed(tl) => write!(f, "{}", tl),
+        }
+    }
+}
+
+impl Display for TypedLit {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TypedLit::Bool(b) => write!(f, "{}", b),
+            TypedLit::Int(i) => write!(f, "{}", i),
+            TypedLit::Float(fl) => write!(f, "{}", fl),
+            TypedLit::Double(d) => write!(f, "{}", d),
+        }
+    }
+}
+
+impl Display for BinaryExpr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            BinaryExpr::Eq(lhs, rhs) => write!(f, "{} == {}", lhs, rhs),
+            BinaryExpr::Ne(lhs, rhs) => write!(f, "{} != {}", lhs, rhs),
+            BinaryExpr::Lt(lhs, rhs) => write!(f, "{} < {}", lhs, rhs),
+            BinaryExpr::Le(lhs, rhs) => write!(f, "{} <= {}", lhs, rhs),
+            BinaryExpr::Gt(lhs, rhs) => write!(f, "{} > {}", lhs, rhs),
+            BinaryExpr::Ge(lhs, rhs) => write!(f, "{} >= {}", lhs, rhs),
+            BinaryExpr::Add(lhs, rhs) => write!(f, "{} + {}", lhs, rhs),
+            BinaryExpr::Sub(lhs, rhs) => write!(f, "{} - {}", lhs, rhs),
+            BinaryExpr::Mul(lhs, rhs) => write!(f, "{} * {}", lhs, rhs),
+            BinaryExpr::Div(lhs, rhs) => write!(f, "{} / {}", lhs, rhs),
+        }
+    }
+}
+
+impl std::fmt::Display for JITType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl std::fmt::Debug for JITType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.code {
+            0 => write!(f, "nil"),
+            0x70 => write!(f, "bool"),
+            0x76 => write!(f, "i8"),
+            0x77 => write!(f, "i16"),
+            0x78 => write!(f, "i32"),
+            0x79 => write!(f, "i64"),
+            0x7b => write!(f, "f32"),
+            0x7c => write!(f, "f64"),
+            0x7e => write!(f, "small_ptr"),
+            0x7f => write!(f, "ptr"),
+            _ => write!(f, "unknown"),
```

Comment:
Hmm, can we match with the defined const? e.g. `BOOL.code`, instead of the actual code value here?

Reply:
It seems I cannot put BOOL.code here. Introducing another const may be too complex?